manifests/claudie/dynamo/dynamodb.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: apps/v1
+kind: Deployment 
+metadata:
+  name: dynamodb
+  labels:
+    app: dynamodb
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dynamodb
+  template:
+    metadata:
+      labels:
+        app: dynamodb
+    spec:
+      volumes:
+        - name: data
+          persistentVolumeClaim:
+            claimName: dynamo-pvc
+      containers:
+      - name: dynamodb
+        volumeMounts:
+            - name: data
+              mountPath: /home/dynamodblocal/data
+        image: amazon/dynamodb-local:latest
+        command:
+          - "java"
+        args:
+          - "-jar"
+          - "DynamoDBLocal.jar"
+          - "-sharedDb"
+          - "-dbPath"
+          - "./data"
+        resources:
+            requests:
+              memory: 512Mi
+              cpu: 100m
+            limits:
+              memory: 1Gi
+              cpu: 200m
+        env:
+          - name: AWS_DEFAULT_REGION
+            value: "local"
+        ports:
+        - containerPort: 8000
+          protocol: TCP
+          name: dynamodb  
+        securityContext:
+          runAsUser: 0
+          allowPrivilegeEscalation: false
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: dynamo
+spec:
+  selector:
+    app: dynamodb
+  ports:
+  - protocol: TCP
+    port: 8000
+    targetPort: 8000
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: dynamo-pvc
+spec:
+  resources:
+    requests:
+      storage: 10Gi
```

Comment:
@MarioUhrik 10Gi looks like a bit of overkill, should we have a lower value, something like 500Mi ?

Reply:
Let's go for 1 Gi, but please make a little bit of a research to verify whether this is enough.

My unproven theory is that the only reason why DynamoDB has to exist, is because it may have a feature that allows atomic writes/reads. So perhaps, in theory, even a nearly empty file would suffice.

Reply:
After doing some rough calculations, we can have more than 2M clusters in 1Gi of storage.

We create two string items with each `terrafom init` command(LockID and Digest), The combined size of both the string field is 200 bytes and since a cluster with LB have 2 `terraform init` executions (one for k8s nodepools and other for LB nodepools) so that make the 400 bytes of data for a cluster with  LB attached. Adding a buffer of 100 bytes for each cluster brings up the total to 500 bytes or half a kilobyte. Since there are a Million KB in one GB, we can have 2 Million entries in the table

Reply:
Thanks for the info :)
I think 1Gi is fine. It's near-nothing. The sadder thing is that some storageclasses have a minimum permissible capacity per PV much higher than that, e.g. I've seen OCI block volumes have a minimum of 50Gi size, even if you only request 1Gi.